server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -3393,6 +3392,20 @@ assert getLocalCheckpoint() == primaryContext.getCheckpointStates().get(routingE
         synchronized (mutex) {
             replicationTracker.activateWithPrimaryContext(primaryContext); // make changes to primaryMode flag only under mutex
         }
+        postActivatePrimaryMode();
+    }
+
+    private void postActivatePrimaryMode() {
+        if (indexSettings.isRemoteStoreEnabled()) {
+            // We make sure to upload translog (even if it does not contain any operations) to remote translog.
+            // This helps to get a consistent state in remote store where both remote segment store and remote
+            // translog contains data.
+            try {
+                getEngine().translogManager().syncTranslog();
+            } catch (IOException e) {
+                throw new IllegalStateException("Failed to sync translog to remote from new primary", e);
```

Comment:
What would be the behaviour due to throwing this exception now during transient remote issues?

Reply:
Lets add an error log for now? We can have create empty translog parameter in remote store api to handle no translog cases.